web/src/components/categories-filter/categories-filter.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+
+import { useState } from "react"
+import { useRouter } from "next/navigation"
+import ISSUE_CATEGORY_LABELS from "@/constants/issue-category-labels"
+import CategoryLabel from "@/components/category-label"
+import updateCurrentPath from "@/utils/update-current-path"
+import type { CategoriesFilterProps } from './types'
+
+const FILTER_OPTIONS = [ { name: "all", label: "All" }, ...ISSUE_CATEGORY_LABELS ]
+
+export default function CategoriesFilter({ filterParams }: CategoriesFilterProps) {
+  const [selection, setSelection] = useState<Set<string>>(new Set(filterParams.category ? filterParams.category.split(',') : ["all"]))
+  const router = useRouter()
+
+  const handleSelectionChange = (name: string) => {
+    const updatedSelection = new Set(updateSelection(name))
+    updatedSelection.delete("all")
+
+    const newPath = updateCurrentPath({ category: Array.from(updatedSelection).join(",") })
+    router.push(newPath)
+  }
```

Comment:
so 

- if I click a category:
  - 'All' will be unslected
  - clicking another category this last will be added to the filters
  - selecting all categories will transform the selection to 'All'

- If I have, for example 2 of 5 filteres, selected, and click 'All', the selected filters will be deleted, and keep 'All' selected

- if I don't have any filter selected, and I click 'All', 'All' won't be deleted


right?